### PR TITLE
[EGGO-42] Support arbitrary Hadoop filesystem for output.

### PR DESCRIPTION
**Pull request by tomwhite**
Here's a first go at this. Most of the changes are just renames to remove "s3" from identifiers.

I tested by changing the URLs in config.py to file:/// URLs and observing that the files were created on the local filesystem. It would be good to parameterize this. @laserson, any thoughts on a good way to do this?

### CODE REVIEW

**Comment by tomwhite** [comment]
I opted to use an env variable EGGO_BASE_URL to control where the datasets go. It defaults to s3n://bdg-eggo.

I've tested this with success locally and on EC2 using the test dataset and I think it's good to go.

File: eggo/dag.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ class DownloadDatasetTask(Task):
 
     def requires(self):
         for source in ToastConfig().config['sources']:
-            dest_name = build_s3_filename(source['url'],
+            dest_name = build_dest_filename(source['url'],
                                           decompress=source['compression'])
```
Comment (laserson) [inline eggo/dag.py:145]: nit: align argument (we'll probably enforce pep8 at some point)

File: eggo/dag.py [hunk 2]
```diff
@@ -205,19 +205,22 @@ def job_runner(self):
 
     def mapper(self, line):
         source = json.loads('\t'.join(line.split('\t')[1:]))
-        dest_name = build_s3_filename(source['url'],
+        dest_name = build_dest_filename(source['url'],
                                       decompress=source['compression'])
```
Comment (laserson) [inline eggo/dag.py:209]: same nit

File: eggo/dag.py [hunk 3]
```diff
@@ -30,31 +30,26 @@
...
 
 
-def target_s3n_url(dataset_name, format='bdg', edition='basic'):
-    return os.path.join(EGGO_S3N_BUCKET_URL, dataset_name, format, edition) + '/'
-
-
-def dataset_s3n_url(dataset_name):
-    return os.path.join(EGGO_S3N_BUCKET_URL, dataset_name) + '/'
-
+def flagTarget(path):
```
Comment (laserson) [inline eggo/dag.py:49]: nit: no camel case in python fns

**Comment by laserson** [comment]
looks great, @tomwhite!

**Comment by laserson** [comment]
Fixed nits and manually merged in 57b2f57fcc476981b9fd43ea02821a919bb456d4.  Thanks @tomwhite!

**Pull request closed without merging**